scripts/qemu-dm-wrapper

```diff
@@ -104,10 +109,21 @@ def main(argv):
     if os.path.exists("/usr/share/qemu-linaro/qemu"):
         qemu_args = qemu_args + ["-L", "/usr/share/qemu-linaro/qemu"]
 
+    if xenstore_exists("/local/domain/%d/platform/nic_type" % domid):
+        print "No explicit NIC selected. Defaulting to rtl8139. "
+    else:
+        nic = xenstore_read("/local/domain/%d/platform/nic_type" % domid)
```

Comment:
Then fix the existing helper not to do that.

Using xenstore-exists is an extra process exec, and a TOCTOU race which can still cause xenstore_read() to raise said exception